src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -302,11 +298,39 @@ public String getIexecUploadToken() {
         // get challenge
         Integer chainId = blockchainAdapterConfigurationService.getChainId();
         try {
-            Eip712Challenge eip712Challenge = resultProxyClient.getChallenge(chainId);
+            final EIP712Challenge eip712Challenge = resultProxyClient.getChallenge(chainId);
+            if (eip712Challenge == null) {
+                log.error("Couldn't retrieve an EIP712Challenge from Result Proxy");
+                return "";
+            }
+
+            final EIP712Domain domain = eip712Challenge.getDomain();
+            if (domain == null) {
+                log.error("Couldn't get a correct domain from EIP712Challenge " +
+                        "retrieved from Result Proxy [eip712Challenge:{}]",
+                        eip712Challenge);
+                return "";
+            }
+
+            final String expectedDomainName = "iExec Core";
```

Comment:
Do we want a generic domain name here or the app name (e.g. `iExec Result Proxy`) to validate the application which emitted the challenge ?

Reply:
Right, see:
- https://github.com/iExecBlockchainComputing/iexec-worker/pull/453/commits/6ffd3e8d3229f62f48b23cde084b923c21990e97
- https://github.com/iExecBlockchainComputing/iexec-core/pull/513/commits/e967e65888a0ada6432830730a36abaf4a47c80e
- https://github.com/iExecBlockchainComputing/iexec-result-proxy/pull/53/commits/5eedf0da577379ac82f83f9287116654b8090d69